### PR TITLE
readding babel for edge >= 18 due to missing support for spread syntax

### PR DESCRIPTION
**Pull request by ktmatzdorf**
i readded babel and introduced a new grunt task called edge which you can run after build like this git pull && npm install --develop && grunt edge && pm2 restart enketo

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -88,6 +88,12 @@ module.exports = grunt => {
             }
         },
         shell: {
+            ie11polyfill: {
```
Comment (MartijnR) [inline Gruntfile.js:91]: not needed

File: Gruntfile.js [hunk 2]
```diff
@@ -107,19 +113,43 @@ module.exports = grunt => {
             },
             rollup: {
                 command: 'npx rollup --config'
+            },
+            ie11polyfill: {
```
Comment (MartijnR) [inline Gruntfile.js:117]: not needed

File: Gruntfile.js [hunk 3]
```diff
@@ -107,19 +113,43 @@ module.exports = grunt => {
...
+                    'mkdir -p public/js/build && curl "https://polyfill.io/v3/polyfill.min.js?ua=ie%2F11.0.0&features=es2015%2Ces2016%2Ces2017%2Ces2018%2Cdefault-3.6%2Cfetch%2CNodeList.prototype.forEach" -o "public/js/build/ie11-polyfill.min.js"',
+                    'cp -f node_modules/enketo-core/src/js/obscure-ie11-polyfills.js public/js/build/obscure-ie11-polyfills.js'
+                ].join( '&&' )
+            },
+            babel: {
+                command: bundles
+                    .map( bundle => `npx babel ${bundle} --out-file ${bundle.replace('-bundle.', '-bundle.')}` )
+                    .join( '&&' )
+            },
+            browserify: {
```
Comment (MartijnR) [inline Gruntfile.js:128]: not needed, right?

File: Gruntfile.js [hunk 4]
```diff
@@ -107,19 +113,43 @@ module.exports = grunt => {
...
+                command: bundles
+                    .map( bundle => `npx browserify node_modules/enketo-core/src/js/workarounds-ie11.js ${bundle.replace('-bundle.', '-bundle.')} -o ${bundle.replace('-bundle.', '-bundle.')}` )
+                    .concat( [ 'rm -f public/js/build/*ie11-temp-bundle.js' ] )
+                    .join( '&&' )
             }
         },
-        eslint: {
-            check: {
-                src: JS_INCLUDE
+        jsbeautifier: {
```
Comment (MartijnR) [inline Gruntfile.js:135]: Why is this back?

File: Gruntfile.js [hunk 5]
```diff
@@ -235,31 +265,32 @@ module.exports = grunt => {
         let content = PRE + paths.map( p => {
             if ( grunt.file.exists( WIDGETS_JS_LOC, `${p}.js` ) ) {
                 num++;
-
                 return `import w${num} from '${p}';`;
             } else {
                 return `//${p} not found`;
             }
-        } ).join( '\n' ) + `\n\nexport default [${[ ...Array( num ).keys() ].map( n => 'w' + ( n + 1 ) ).join( ', ' )}];`;
+        } ).join( '\n' ) + `\n\nexport default [${[...Array(num).keys()].map(n => 'w'+(n+1)).join(', ')}];`;
```
Comment (MartijnR) [inline Gruntfile.js:272]: style degradation

File: babel.config.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const config = {
+    "presets": [
+        [
+            "@babel/env",
+            {
+                "targets": {
+                    "edge": "18",
+                    "firefox": "60",
```
Comment (MartijnR) [inline babel.config.js:8]: Why firefox too?

File: babel.config.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+                }
+            }
+        ]
+    ],
+    plugins: [
+        [require("@babel/plugin-proposal-class-properties"), { loose: true }],
```
Comment (MartijnR) [inline babel.config.js:14]: necessary?

File: package.json [hunk 8]
```diff
@@ -88,6 +89,12 @@
         "xml2js": "^0.4.23"
     },
     "devDependencies": {
+        "@babel/cli": "^7.10.1",
+        "@babel/core": "^7.10.1",
+        "@babel/plugin-syntax-dynamic-import": "^7.8.3",
```
Comment (MartijnR) [inline package.json:94]: is this necessary?

**Comment by ktmatzdorf** [comment]
@MartijnR i cleaned the commit, you were right i did not need a lot of stuff i reintroduced. my major issue was that i was using the wrong babel version (6 instead of 7). that was fixed by uninstalling babel `npm uninstall babel` and reinstalling @babel/core `npm --save-dev install @babel/core @babel/cli @babel/plugin-transform-spread @babel/preset-env`

**Comment by MartijnR** [comment]
Thanks @ktmatzdorf. Just wanted to let you know I have not forgotten. Am just a little busy at the moment. Will check this out as soon as I can.

**Comment by MartijnR** [comment]
Thanks! One tip for a future PR is always create a separate branch.

**Pull request merged**